### PR TITLE
RTEMIS-78: Fixed Caching issue & Permissions issue.

### PR DESCRIPTION
**Pull request by RohanSingh-inno**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-78

---
### What the PR does
`(Explain fixes, changes and impacts)`
- District user can add school under his district.
- Fixed permission issue for state and district admin.
- Fixed caching issue for the pending request panel.

### CODE REVIEW

File: modules/rte_mis_state/rte_mis_state.module [hunk 1]
```diff
@@ -557,3 +557,16 @@ function rte_mis_state_entity_operation_alter(array &$operations, EntityInterfac
...
 }
+
+/**
+ * Implements hook_user_presave().
+ */
+function rte_mis_state_user_presave(UserInterface $account) {
+  // Check if the current user has the district admin role.
+  if ($account->hasRole('district_admin')) {
+    // Invalidate cache tag when user data is updated.
+    $cache_tags = ['taxonomy_term_list'];
```
Comment (kunal-sin) [inline modules/rte_mis_state/rte_mis_state.module:568]: Please be specific to school vocab.

Comment (RohanSingh-inno) [reply modules/rte_mis_state/rte_mis_state.module]: ok

File: modules/rte_mis_state/rte_mis_state.module [hunk 2]
```diff
@@ -557,3 +557,27 @@ function rte_mis_state_entity_operation_alter(array &$operations, EntityInterfac
...
+/**
+ * Implements hook_user_presave().
+ */
+function rte_mis_state_user_presave(UserInterface $account) {
+  // Check if the current user has the district admin role.
+  if ($account->hasRole('district_admin')) {
+    // Invalidate cache tag when user data is updated.
+    $cache_tags = ['taxonomy_term_list:school'];
+    Cache::invalidateTags($cache_tags);
+  }
```
Comment (kunal-sin) [inline modules/rte_mis_state/rte_mis_state.module:581]: Did we validated invalidation without this code?

Comment (RohanSingh-inno) [reply modules/rte_mis_state/rte_mis_state.module:581]: yes, the cache tag was not invalidated by default. hence, I had to invalidate it manually.

**Pull request merged**